modules/cert-manager-acme-dns01-aws.adoc

```diff
@@ -0,0 +1,113 @@
+
+.Prerequisites
+
+* You have access to the {product-title} cluster as a user with the `cluster-admin` role.
+* You have set up an IAM role for Amazon Route 53. For more information, see link:https://cert-manager.io/docs/configuration/acme/dns01/route53/[Route53] in the upstream cert-manager documentation.
++
+[NOTE]
+====
+For the {cert-manager-operator}, you must provide an `accessKeyID` and `secretAccessKey` to specify the credentials.
+====
+
+.Procedure
+
+. Configure the nameserver settings for the DNS01 self check:
+
+.. Edit the `CertManager` resource by running the following command:
++
+[source,terminal]
+----
+$ oc edit certmanager cluster
+----
+
+.. Add a `spec.controllerConfig` section with the following override arguments:
++
+[source,yaml]
+----
+apiVersion: operator.openshift.io/v1alpha1
+kind: CertManager
+metadata:
+  name: cluster
+  ...
+spec:
+  ...
+  controllerConfig:                                <1>
+    overrideArgs:
+      - '--dns01-recursive-nameservers=1.1.1.1:53' <2>
+      - '--dns01-recursive-nameservers-only'       <3>
+----
+<1> Add the `spec.controllerConfig` section.
+<2> Provide a comma-separated list of `<host>:<port>` nameservers to query for the DNS01 self check.
+<3> Specify to only use recursive nameservers.
++
+// TODO: what exactly does this mean? That it uses caching instead of checking the authoritative nameservers every time? Might cause it to take longer it says why? (every time)?
++
+// TODO: check if v1alpha1 is the correct version?
++
+// TODO Find out if that is just an example value, where they need to replace?
+
+.. Save the file to apply the changes.
+
+. Create an issuer.
+
+.. Create a YAML file that defines the `Issuer` object:
++
+.Example `issuer.yaml` file
+[source,yaml]
+----
+  apiVersion: cert-manager.io/v1
+  kind: Issuer
+  metadata:
+    name: letsencrypt-staging                                        <1>
+    namespace: sandbox                                               <2>
+  spec:
+    acme:
+      server: https://acme-staging-v02.api.letsencrypt.org/directory <3>
+      email: "thn@redhat.com"                                        <4>
+      privateKeySecretRef:
+        name: letsencrypt-staging                                    <5>
+      solvers:
+      - dns01:
+          route53:
+            accessKeyID: AKIAT5TK4ZONNAZQMGOD                        <6>
+            hostedZoneID: Z3URY6TWQ91KVV                             <7>
+            region: us-east-1
+            secretAccessKeySecretRef:
+              name: "aws-secret"                                     <8>
+              key: "awsSecretAccessKey"                              <9>
+        selector:
+          dnsNames:
+          - 'example.org'                                            <10>
+----
+<1> TODO
+<2> Specify the namespace for the CA issuer.
+<3> TODO
+<4> #❗Replace this with your own email address
+<5> TODO
```

Comment:
@thejasn @swghosh What is this `privateKeySecretRef` of `letsencrypt-staging` referencing? I don't see this anywhere else in the procedure. I need to know what to say that this should be refereencing

Reply:
> 	privateKeySecretRef is the name of a Kubernetes Secret resource that will be used to store the automatically generated ACME account private key. Optionally, a key may be specified to select a specific entry within the named Secret resource. If key is not specified, a default of tls.key will be used.

https://cert-manager.io/docs/reference/api-docs/#acme.cert-manager.io/v1.ACMEIssuer

Reply:
@thejasn I'm assuming that his has to exist already then. Should we add a step to add this secret? Or just say that it has to already exist?

Reply:
Works both ways, we can recommend to rely it creating a new secret rather than update an existing secret (at least for a fresh setup).